### ISSUE TITLE
SimCLR fixed to a dataset

### ISSUE THREAD

**Issue by lbugnon** [PyTorchLightning/lightning-bolts#555]
Why  is SimCLR fixed to some datasets now? I tried to run a code from the last year but now it requires to set a dataset like cifar. I want to use my own data. Is this a bug or a feature?
Thanks!

**Comment by akihironitta** [comment]
@lbugnon If you look at the current implementation in the codebase, you'll find out that we only have configurations for stl10, cifar10, and imagenet, so it's not a bug. I'm sure you can use SimCLR for your own dataset similarly to the below datasets:
https://github.com/PyTorchLightning/pytorch-lightning-bolts/blob/86e3f52308fa02e2a988b4977858a945f7d15ab2/pl_bolts/models/self_supervised/simclr/simclr_module.py#L348
https://github.com/PyTorchLightning/pytorch-lightning-bolts/blob/86e3f52308fa02e2a988b4977858a945f7d15ab2/pl_bolts/models/self_supervised/simclr/simclr_module.py#L363
https://github.com/PyTorchLightning/pytorch-lightning-bolts/blob/86e3f52308fa02e2a988b4977858a945f7d15ab2/pl_bolts/models/self_supervised/simclr/simclr_module.py#L383

**Comment by lbugnon** [comment]
Thanks, I've read the code. The question is why to use a fixed dataset in
the model definition. In previous version, I can use the model and the
lightning trainer with any custom dataset, like all the other models. Now,
I have to redefine the SimCLR class for each new dataset I use, or there is
something I'm failing to see.

**Comment by akihironitta** [comment]
@ananyahjha93 Would you mind answering this question?

**Comment by ananyahjha93** [comment]
@lbugnon The only reason for having dataset within the model is to accommodate the labelled and unlabelled parts of stl-10. 

https://github.com/PyTorchLightning/pytorch-lightning-bolts/blob/master/pl_bolts/models/self_supervised/simclr/simclr_module.py#L160

Otherwise there is no need to add dataset to the model directly. This will be corrected in a new release. But, since we need to compute per step linear warmup and cosine decay learning rates, we do need to know the number of samples in the dataset.

**Comment by jmarrietar** [comment]
Hi everyone!  My 0.2 cents: 

+1 to what @lbugnon says. Maybe this implementation should be more flexible and general to custom datasets ?.
